public/app/plugins/datasource/prometheus/components/VariableQueryEditor.tsx

```diff
@@ -0,0 +1,257 @@
+      qryType: qryType,
+      label,
+      metric,
+      varQuery,
+      seriesQuery,
+      refId: 'PrometheusVariableQueryEditor-VariableQuery',
+    };
+
+    const queryString = migrateVariableEditorBackToVariableSupport(queryVar);
+
+    onChange({
+      query: queryString,
+      refId,
+    });
+  };
+
+  const onQueryTypeChange = (newType: SelectableValue<QueryType>) => {
+    setQryType(newType.value);
+    if (newType.value === QueryType.LabelNames) {
+      onChangeWithVariableString(newType.value);
+    }
+  };
+
+  const onLabelChange = (newLabel: SelectableValue<string>) => {
+    setLabel(newLabel.value ?? '');
+  };
+
+  const onMetricChange = (e: FormEvent<HTMLInputElement>) => {
+    setMetric(e.currentTarget.value);
+  };
+
+  const onVarQueryChange = (e: FormEvent<HTMLTextAreaElement>) => {
+    setVarQuery(e.currentTarget.value);
+  };
+
+  const onSeriesQueryChange = (e: FormEvent<HTMLInputElement>) => {
+    setSeriesQuery(e.currentTarget.value);
+  };
+
+  const handleBlur = () => {
+    if (qryType === QueryType.LabelNames) {
+      onChangeWithVariableString(qryType);
+    } else if (qryType === QueryType.LabelValues && label) {
+      onChangeWithVariableString(qryType);
+    } else if (qryType === QueryType.MetricNames && metric) {
+      onChangeWithVariableString(qryType);
+    } else if (qryType === QueryType.VarQueryResult && varQuery) {
+      onChangeWithVariableString(qryType);
+    } else if (qryType === QueryType.SeriesQuery && seriesQuery) {
+      onChangeWithVariableString(qryType);
+    }
+  };
+
+  return (
+    <InlineFieldRow>
+      <InlineField
+        label="Query Type"
+        labelWidth={20}
+        tooltip={
+          <div>The Prometheus data source plugin provides the following query types for template variables.</div>
+        }
+      >
+        <Select
+          placeholder="Select query type"
+          aria-label="Query type"
+          onChange={onQueryTypeChange}
+          onBlur={handleBlur}
+          value={qryType}
+          options={variableOptions}
+          width={25}
+        />
+      </InlineField>
+      {qryType === QueryType.LabelValues && (
+        <>
+          <InlineField
+            label="Label"
+            labelWidth={20}
+            required
+            tooltip={
+              <div>
+                Returns a list of label values for the label name in all metrics unless the metric is specified.
+              </div>
+            }
+          >
+            <Select
+              aria-label="label-select"
+              onChange={onLabelChange}
+              onBlur={handleBlur}
+              value={label}
+              options={labelOptions}
+              width={25}
+              allowCustomValue
+            />
+          </InlineField>
+          <InlineField
+            label="Metric"
+            labelWidth={20}
+            tooltip={<div>*Optional: returns a list of label values for the label name in the specified metric.</div>}
```

Comment:
```suggestion
            tooltip={<div>Optional: returns a list of label values for the label name in the specified metric.</div>}
```

Reply:
Just that we're using the asterisk to denote required fields, the asterisk in this context is kinda confusing